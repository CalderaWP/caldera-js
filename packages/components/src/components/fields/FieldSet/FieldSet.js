import { fieldSetClassNames, parseAttributes } from '../util';
import PropTypes from 'prop-types';
import React from 'react';

/**
 *
 * @param fieldType
 * @param attributes
 * @param children
 * @return {*}
 * @constructor
 */
export const FieldSet = ({ fieldType, attributes, children, legend }) => {
	attributes = parseAttributes(attributes, 'fieldset');
	return (
		<fieldset className={fieldSetClassNames(fieldType)} {...attributes}>
			<legend>{legend}</legend>
			{children}
		</fieldset>
	);
};

FieldSet.propTypes = {
	fieldType: PropTypes.string.isRequired,
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node
	]).isRequired,
	attributes: PropTypes.object
};
