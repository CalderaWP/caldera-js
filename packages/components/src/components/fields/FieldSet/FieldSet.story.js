import React from 'react';
import { storiesOf } from '@storybook/react';
import { FieldSet } from './FieldSet';
import { InputField } from '../InputField/InputField';
const onChange = newValue => {};
storiesOf('FieldSet', module).add('Wraps an input', () => (
	<FieldSet fieldType={'text'}>
		<input />
	</FieldSet>
));

storiesOf('FieldSet', module).add('With checkboxes inside', () => (
	<FieldSet fieldType={'text'} legend={'Checkbox Fieldset'}>
		<InputField
			label={'Not Checked'}
			html5type={'checkbox'}
			value={false}
			onChange={onChange}
		/>
		<InputField
			label={'Checked'}
			html5type={'checkbox'}
			value={true}
			onChange={onChange}
		/>
	</FieldSet>
));
