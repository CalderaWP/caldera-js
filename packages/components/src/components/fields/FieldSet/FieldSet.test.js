import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { FieldSet } from './FieldSet';

describe('Field label', () => {
	it('Matches snapshot', () => {
		const component = renderer.create(
			<FieldSet fieldType={'checkbox'} legend={'Check All options'}>
				<label htmlFor="one">One</label>
				<input id="one" type={'checkbox'} />
				<label htmlFor="two">Two</label>
				<input id="two" type={'checkbox'} />
			</FieldSet>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
	it('Allows attributes', () => {
		const attributes = {
			form: 'contactForm',
			disable: true,
			name: 'roys'
		};
		const component = mount(
			<FieldSet
				fieldType={'checkbox'}
				legend={'Check All options'}
				attributes={attributes}
			>
				<label htmlFor="one">One</label>
				<input id="one" type={'checkbox'} />
				<label htmlFor="two">Two</label>
				<input id="two" type={'checkbox'} />
			</FieldSet>
		);
		expect(component.find('fieldset').prop('form')).toEqual(
			attributes.form
		);
	});
});
