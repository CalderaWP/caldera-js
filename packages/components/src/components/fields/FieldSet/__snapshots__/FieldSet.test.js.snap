// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field label Matches snapshot 1`] = `
<fieldset
  className="caldera-fieldset caldera-fieldset-checkbox"
>
  <legend>
    Check All options
  </legend>
  <label
    htmlFor="one"
  >
    One
  </label>
  <input
    id="one"
    type="checkbox"
  />
  <label
    htmlFor="two"
  >
    Two
  </label>
  <input
    id="two"
    type="checkbox"
  />
</fieldset>
`;
